### PR TITLE
Add onFocus command suggestion list [++, ui]

### LINKED ISSUE #93: Add onFocus command suggestion list [++, ui]

### LINKED ISSUE #94: Change commands and text to follow mockup [ui]

### LINKED ISSUE #95: Filter suggestion list as input changes [++, logic]

### PR DESCRIPTION
**Pull request by Jillzyt**
Add autocomplete

### CODE REVIEW

**Comment by Jillzyt** [comment]
Currently, the autocomplete checks if the first three letters are similar and suggest the commands' format. 

How I implemented autocomplete:
- Add getSuggestionList() in LogicManager so that I can retrieve the list of command suggestions from the commands. (UI will only have access to logic) 
- Add CommandBox's commandTextField setListenerForAutoComplete() method set listener to detect changes in text field and sends input to showAutoCompleteResult().
- Add showAutoCompleteResult() in ResultDisplay so that we show results to the user.

**Comment by Jillzyt** [comment]
image
image
image

File: src/main/java/seedu/fma/ui/ResultDisplay.java [hunk 1]
```diff
@@ -25,4 +30,28 @@ public void setFeedbackToUser(String feedbackToUser) {
...
+     */
+    public void showAutoCompleteResult(String input) {
+        requireNonNull(input);
+
+        if (input.length() == 0) {
+            return;
+        }
+
+        String feedback = "";
+        input = input.length() > 3 ? input.substring(0, 3) : input;
```
Comment (therizhao) [inline src/main/java/seedu/fma/ui/ResultDisplay.java]: Could split input and use the first string
`input.split(' ')[0]`
-> checking the first 3 char substring may be brittle (e.g. may cause unforeseen bugs if we change the commands e.g. if got addelephant command, addex will show addex and addelephant)

**Review by therizhao — CHANGES_REQUESTED** [review]
1 minor issue

File: src/main/java/seedu/fma/ui/ResultDisplay.java (hunk 1 above)
Comment (Jillzyt) [reply src/main/java/seedu/fma/ui/ResultDisplay.java]: Resolved.

**Review by therizhao — APPROVED** [review]
nice

**Pull request merged**